src/get-started/java-binding.md

```diff
@@ -395,6 +466,62 @@ at `/api/services/cryptocurrency/balance/:walletId`.
 See [documentation][vertx-web-docs] on the possibilities of `Vert.x` used as a web
 framework.
 
+### Configuration
+
+#### Initialization and Initial Configuration
+
+If a service needs to initialize its persistent state when it is started,
+potentially, using some configuration parameters, it shall
+do that in [`Service#initialize`][service-initialize].
+
+<!-- TODO: Link to the sections on starting new services -->
+
+Exonum invokes this method when a new service instance is added
+to the blockchain. Administrators, initiating this action,
+may pass some initialization parameters. A service is responsible for 
+validation of these parameters. If they are incorrect, it shall throw
+an Exception, which will abort the start of this service instance.
+ 
+A service shall also update its _persistent_ state based on these parameters.
+It shall not derive the `Service` _object_ state from them, as 
+Exonum may instantiate multiple `Service` objects for a single 
+service instance (e.g., when a node is stopped and then restarted).
+
+For example, a service may set some initial values in its collections, 
+or save all or some configuration parameters as is for later retrieval in transactions
+and/or read requests.
+
+<!-- TODO: example -->
+
+#### Reconfiguration
+
+<!-- TODO: link the complete documentation on reconfiguration,
+especially in terms of its invocation by administrators -->
+
+Exonum supervisor service provides a mechanism to reconfigure
+the started service instances. The re-configuration protocol for _services_
+is similar to the one for consensus configuration. 
+This protocol includes the following steps:
+1. a proposal of a new configuration; 
+2. verification of its correctness; 
+3. approval of the proposal; and
+4. application of the new approved configuration.
+
+The protocol of the _proposal_ and _approval_ steps is determined by the installed 
+supervisor service.
+The service must implement _verification_ and _application_ of the parameters by
+implementing [`Configurable`][configurable] interface.
```

Comment:
Add links for [configurable] and [standard-supervisor-rustdoc] (or corresponding TODOs)?